components/url_sanitizer/browser/url_sanitizer_service_unittest.cc

```diff
@@ -0,0 +1,143 @@
+#include "base/test/task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+namespace {
+const char kTestPatterns[] = R"(
+[
+  {
+    "include": [
+      "*://*.twitter.com/*"
+    ],
+    "params": [
+      "t"
+    ]
+  },
+  {
+    "include": [
+      "*://*/*"
+    ],
+    "exclude": [
+    ],
+    "params": [
+      "utm_content",
+      "utm_affiliate"
+    ]
+  },
+  {
+    "include": [
+      "https://dev-pages.bravesoftware.com/clean-urls/*"
+    ],
+    "exclude": [
+      "https://dev-pages.bravesoftware.com/clean-urls/exempted/*"
+    ],
+    "params": [
+      "brave_testing1",
+      "brave_testing2"
+    ]
+  }
+])";
+
+}  // namespace
+
+class URLSanitizerServiceUnitTest : public testing::Test,
+                                    public URLSanitizerService {
+ public:
+  URLSanitizerServiceUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {}
+
+  void WaitInitialization(const std::string& json) {
+    base::RunLoop loop;
+    SetInitializationCallbackForTesting(loop.QuitClosure());
+    Initialize(json);
+    loop.Run();
+  }
+
+ private:
+  base::test::TaskEnvironment task_environment_;
+};
+
+TEST_F(URLSanitizerServiceUnitTest, StripQueryParameter) {
+  auto list = base::MakeFlatSet<std::string>(
+      std::vector<std::string>{"fbclid", "second"});
+
+  EXPECT_EQ(StripQueryParameter("fbclid=11&param1=1&second=2", list),
+            "param1=1");
+  EXPECT_EQ(
+      StripQueryParameter(
+          "fbclid=11&fbclid=11&fbclid=22&param1=1&second=2&second=2&second=2",
+          list),
+      "param1=1");
+  EXPECT_EQ(StripQueryParameter("param1=1", list), "param1=1");
+  EXPECT_EQ(StripQueryParameter("", list), "");
+}
+
+TEST_F(URLSanitizerServiceUnitTest, ClearURLS) {
+  // The service has not yet been initialized.
+  EXPECT_FALSE(SanitizeURL(GURL("https://brave.com")).is_valid());
+  WaitInitialization(kTestPatterns);
+
+  EXPECT_EQ(
+      SanitizeURL(
+          GURL("https://dev-pages.bravesoftware.com/clean-urls/"
+               "exempted/"
+               "?brave_testing1=foo&brave_testing2=bar&brave_testing3=keep&&;b&"
+               "d&utm_content=removethis&e=&f=g&=end")),
+      GURL("https://dev-pages.bravesoftware.com/clean-urls/"
+           "exempted/?brave_testing1=foo&brave_testing2=bar&brave_"
+           "testing3=keep&&;b&d&e=&f=g&=end"));
+  EXPECT_EQ(
+      SanitizeURL(GURL(
+          "https://dev-pages.bravesoftware.com/clean-urls/"
+          "?brave_testing1=foo&brave_testing2=bar&brave_testing3=keep&&;b&d&"
+          "utm_content=removethis&e=&f=g&=end")),
+      GURL("https://dev-pages.bravesoftware.com/clean-urls/"
+           "?brave_testing3=keep&&;b&d&e=&f=g&=end"));
+
+  WaitInitialization(R"([
+    { "include": [ "*://*/*"], "params": ["query"] }
+  ])");
+
+  EXPECT_EQ(SanitizeURL(GURL("https://brave.com/?query=removethis")),
+            GURL("https://brave.com/?"));
```

Comment:
In this case we should ideally delete the query string and end up with just `https://brave.com/` (no question mark at the end).

Reply:
done